src/NUnitFramework/framework/Attributes/RandomAttribute.cs

```diff
@@ -33,94 +35,542 @@ namespace NUnit.Framework
+        }
+
+        /// <summary>
+        /// Construct a set of doubles within a specified range
         /// </summary>
-        /// <param name="min"></param>
-        /// <param name="max"></param>
-        /// <param name="count"></param>
         public RandomAttribute(double min, double max, int count)
         {
-            this.count = count;
-            this.dmin = min;
-            this.dmax = max;
-            this.sampleType = SampleType.DoubleRange;
+            _source = new DoubleDataSource(min, max, count);
         }
 
         /// <summary>
-        /// Construct a set of ints from min to max
+        /// Construct a set of floats within a specified range
         /// </summary>
-        /// <param name="min"></param>
-        /// <param name="max"></param>
-        /// <param name="count"></param>
-        public RandomAttribute(int min, int max, int count)
+        public RandomAttribute(float min, float max, int count)
+        {
+            _source = new FloatDataSource(min, max, count);
+        }
+
+        /// <summary>
+        /// Construct a set of bytes within a specified range
+        /// </summary>
+        public RandomAttribute(byte min, byte max, int count)
         {
-            this.count = count;
-            this.min = min;
-            this.max = max;
-            this.sampleType = SampleType.IntRange;
+            _source = new ByteDataSource(min, max, count);
         }
 
         /// <summary>
-        /// Get the collection of _values to be used as arguments
+        /// Construct a set of sbytes within a specified range
         /// </summary>
-        public new IEnumerable GetData(ParameterInfo parameter)
-        {
-            Randomizer r = Randomizer.GetRandomizer(parameter);
-            IList values;
-
-            switch (sampleType)
-            {
-                default:
-                case SampleType.Raw:
-                    if (parameter.ParameterType.IsEnum)
-                        values = r.GetEnums(count, parameter.ParameterType);
-                    else
-                    values = r.GetDoubles(count);
-                    break;
-                case SampleType.IntRange:
-                    values = r.GetInts(min, max, count);
-                    break;
-                case SampleType.DoubleRange:
-                    values = r.GetDoubles(dmin, dmax, count);
-                    break;
-            }
-
-            // Copy the random _values into the data array
-            // and call the base class which may need to
-            // convert them to another type.
-            this.data = new object[values.Count];
-            for (int i = 0; i < values.Count; i++)
-                this.data[i] = values[i];
- 
-            return base.GetData(parameter);
+        [CLSCompliant(false)]
+        public RandomAttribute(sbyte min, sbyte max, int count)
+        {
+            _source = new SByteDataSource(min, max, count);
+        }
```

Comment:
Funny thing... You can create a decimal constructor but you can't use it except for constructing the attribute programaticaly! Decimals are not allowed as arguments to attribtues applied to methods.

In fact, I had one and then realized it couldn't be used so I took it out so it wouldn't appear in the intellisense.